src/node_report.cc

```diff
@@ -205,11 +205,14 @@ static void WriteNodeReport(Isolate* isolate,
            tm_struct.tm_min,
            tm_struct.tm_sec);
   writer.json_keyvalue("dumpEventTime", timebuf);
-  struct timeval ts;
-  gettimeofday(&ts, nullptr);
-  writer.json_keyvalue("dumpEventTimeStamp",
-                       std::to_string(ts.tv_sec * 1000 + ts.tv_usec / 1000));
 #endif
+
+  uv_timeval_t ts;
+  if (uv_gettimeofday(&ts) == 0) {
+    writer.json_keyvalue("dumpEventTimeStamp",
+                         std::to_string(ts.tv_sec * 1000 + ts.tv_usec / 1000));
```

Comment:
This might overflow when `tv_sec` is of a 32 bits long.

(I kind of regret that we didn't use `int64_t` for `uv_timeval_t.tv_sec`...)

Reply:
Maybe there's still time to change `uv_gettimeofday` to `uint64_t uv_gettimeofday()` (or `unsigned long long int`) and always return microseconds? Seems like `uv_timeval_t` is a little "legacy" for this API

Reply:
That's not a bad idea. Since `uv_gettimeofday()` hasn't been released yet, there is time.

Reply:
We can discuss at https://github.com/libuv/libuv/issues/2243.